airflow/triggers/base.py

```diff
@@ -119,3 +119,34 @@ def __eq__(self, other):
         if isinstance(other, TriggerEvent):
             return other.payload == self.payload
         return False
+
+
+class BaseTaskEndEvent(TriggerEvent):
+    """Base event class to end the task without resuming on worker."""
+
+    task_instance_state: TaskInstanceState
+
+    def __init__(self, *, xcom_return: Any = None, other_xcom: dict[str, Any] | None = None, **kwargs):
```

Comment:
this particular bit of the interface... open to suggestions on.... basically anything in this file there's a lot of ways we could've gone... but in particular the xcom_return vs other_xcom split and naming.

i thought having them explicit like this is better than having something "implicitly" go to `return_value` etc